sktime/forecasting/statsforecast.py

```diff
@@ -552,3 +555,255 @@ def get_test_params(cls, parameter_set="default"):
+    def get_test_params(cls, parameter_set="default"):
+        """Return testing parameter settings for the estimator.
+
+        Parameters
+        ----------
+        parameter_set : str, default="default"
+            Name of the set of test parameters to return, for use in tests. If no
+            special parameters are defined for a value, will return `"default"` set.
+            There are currently no reserved values for forecasters.
+
+        Returns
+        -------
+        params : dict or list of dict, default = {}
+            Parameters to create testing instances of the class
+            Each dict are parameters to construct an "interesting" test instance, i.e.,
+            `MyClass(**params)` or `MyClass(**params[i])` creates a valid test instance.
+            `create_test_instance` uses the first (or only) dictionary in `params`
+        """
+        from sktime.forecasting.theta import ThetaForecaster
+
+        del parameter_set  # to avoid being detected as unused by ``vulture`` etc.
+
+        params = {
+            "season_length": [3, 12],
+            "trend_forecaster": ThetaForecaster(),
+        }
+
+        return params
+
+
+class StatsForecastBackAdapterMSTL(BaseForecaster):
+    """StatsForecast Back Adapter for StatsForecast MSTL.
+
+    StatsForecastBackAdapter is a wrapper for
+
+    Parameters
+    ----------
+    estimator :
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.statsforecast import StatsForecastMSTL
+    >>> from sktime.forecasting.statsforecast import StatsForecastBackAdapterMSTL
+    >>> from sktime.forecasting.theta import ThetaForecaster
+
+    >>> y = load_airline()
+    >>> trend_forecaster = StatsForecastBackAdapterMSTL(ThetaForecaster())
+    >>> model = StatsForecastMSTL(
+            season_length=[3,12],
+            trend_forecaster=trend_forecaster
+        )
+    >>> fitted_model = model.fit(y=y)
+    >>> y_pred = fitted_model.predict(fh=[1,2,3])
+    >>> y_pred
+    1961-01    454.507079
+    1961-02    419.341442
+    1961-03    477.373918
+    Freq: M, Name: Number of airline passengers, dtype: float64
+    """
+    def __init__(self, estimator):
+        self.estimator = estimator
+        self._forecaster = None
+        self._model = None
+        self.in_sample_fh = None
+        self.in_sample_X = None
+
+        super().__init__()
+
+        if hasattr(self.estimator, "seasonal"):
+            if self.estimator.seasonal:
+                raise Exception(
+                    "Trend forecaster should not adjust seasonal models."
+                )
+
+        if hasattr(self.estimator, "sp"):
+            if self.estimator.sp > 1:
+                raise Exception(
+                    "Trend forecaster should not adjust seasonal models. Please pass "
+                    "`sp=1` to your trend forecaster"
+                )
+
+    def __repr__(self):
+        return "StatsForecastBackAdapterMSTL"
```

Comment:
I'm not sure if this is necessary. Can you please share the `repr` output in presence and absence of this?

Reply:
Internally, MSTL tries to access attributes of `trend_forecaster` that won't exist if `trend_forecaster` is a back adapter. Here's a snippet from the StatsForecast source code for MSTL:
```python
# check ETS model doesnt have seasonality
if repr(trend_forecaster) == "AutoETS":
    if trend_forecaster.model[2] != "N":
        raise Exception(
            "Trend forecaster should not adjust " "seasonal models."
        )
```
Since not all estimators have `model`, I thought this would be messy to implement. 
I thought adding `__repr__` would solve the problem, but I understand that it is confusing. Is there a better way to do this?
(BTW, `repr` output in absence of this is: StatsForecastBackAdapterMSTL(estimator=AutoETS()))